### DISCUSSION TITLE
Simplify ERC20 extension

### DISCUSSION THREAD

**Discussion by martriay** [OpenZeppelin/cairo-contracts#103] [Design]
I was thinking that maybe we could abstract ERC20 even further and move `transferFrom`, `increaseAllowance`, etc. implementations to the base contract, and require frontend contracts just to expose them instead of re-implementing them. What do you think?

**Comment by andrew-fleming** [comment]
I was considering creating a second contract for frontend contracts from which to import the external methods since we're rewriting them per instance; however, it would be easier and better if we just held them in the base contract. I only see upside here
